llvm/test/CodeGen/X86/lea-fixup-blockaddress.mir

```diff
@@ -0,0 +1,30 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 3
+# RUN: llc -mtriple=i386-unknown-linux-gnu -mattr=slow-3ops-lea -run-pass x86-fixup-LEAs -o -  %s | FileCheck %s
+
+--- |
+  define i32 @square(i32 %0) local_unnamed_addr {
```

Comment:
I was expecting we have a relatively complete test case in the filed issue, but just looks like nitpick :)

Reply:
Thank you for approving :D

However, what do you mean with complete test case in the issue? 
Is my test case not complete enough or do you mean that I should have attached the testcase to the issue? :D

Reply:
I mean you have said the MIR is not generated by this IR. And I cannot see the problem when compile the IR into assembly. So I assume you have a relative complete test that can show the problem from IR or C level.

Reply:
Ah yeah, I understand, but no, unfortunately, I do not. I have a huge cpp file where I can reproduce the issue, but it is way too big and I'm not allowed to disclose it :/

I was not able to reproduce the issue with anything else than this cpp file. Running LLC and an IR file by itself, did not work. Somewhere in the lowering, the IR code would get transformed differently so that the bug did not occur. 

This is why I crafted this MIR test case :D